src/makePersistable.ts

```diff
@@ -18,5 +18,7 @@ export const makePersistable = <T extends { [key: string]: any }, P extends keyo
 
   PersistStoreMap.set(target, mobxPersistStore);
 
+  await mobxPersistStore.init();
```

Comment:
You don't need call await and return mobxPersistStore on next line. After change init body we can return promise with PersistStore instance. 

```ts
return mobxPersistStore.init()
```